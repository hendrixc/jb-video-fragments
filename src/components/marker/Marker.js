import React, { Component } from 'react';
import './marker.css';
import PropTypes from 'prop-types';

class Marker extends Component {
  render() {
    const { title } = this.props;
    return (
      <div className="Marker">
        <span className="side-container">
          <span className="side-arrow" />
          <span className="side-marker">{title}</span>
        </span>
      </div>
    );
  }
}

Marker.propTypes = {
  title: PropTypes.string,
};

Marker.defaultProps = {
  title: '',
};

export default Marker;
