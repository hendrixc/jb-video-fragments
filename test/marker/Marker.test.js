import React from 'react';
import renderer from 'react-test-renderer';
import Marker from '../../src/components/marker/Marker';

jest.mock('../../src/components/video/Video', () => 'Video');
jest.mock('../../src/components/clips/ClipList', () => 'ClipList');

it('renders an marker component into the App as snapshot', () => {
  const TextInputComponent = renderer.create(<Marker />).toJSON();
  expect(TextInputComponent).toMatchSnapshot();
});
