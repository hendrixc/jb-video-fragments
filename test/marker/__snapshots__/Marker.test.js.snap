// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an marker component into the App as snapshot 1`] = `
<div
  className="Marker"
>
  <span
    className="side-container"
  >
    <span
      className="side-arrow"
    />
    <span
      className="side-marker"
    >
      
    </span>
  </span>
</div>
`;
